### PR TITLE
WIP: Add Java bindings for Android

### PR DESCRIPTION
**Pull request by Meakk**
- [x] Build VTK in CI in 64-bits ARM and x86 architectures
- [x] Build libf3d with minimal modules in CI
- [x] Add JNI wrapper in CI (expose `libf3d-java.so` artifacts)

Will be done later:
- [ ] Create Java or Kotlin interface
- [ ] Create Gradle build system
- [ ] Generate `.apk` (expose in CI artifacts)
- [ ] Add Exodus module
- [ ] Add Assimp module
- [ ] Add OCCT module
- [x] Add 32-bits architectures

### CODE REVIEW

**Comment by Meakk** [comment]
Current status:  

- `VTK` is building fine on arm64 and x64, fixed [here](https://gitlab.kitware.com/vtk/vtk/-/merge_requests/8775) but fails for 32-bits architectures.
- `libf3d` is building properly without OCCT/Assimp modules (it's currently 42MB, we need to strip unused symbols)
- JNI native wrapper is done and working
- VTK multitouch interactor style is really bad, we will need to create one ourselves
- Java interface prototype is done, but will need a complete rewrite (ideally in Kotlin) to create a good user experience (let's discuss the specs)

FYI @mwestphal @jpouderoux 

image

**Comment by mwestphal** [comment]
> Java interface prototype is done, but will need a complete rewrite (ideally in Kotlin) to create a good user experience (let's discuss the specs)

What does this mean ?

**Comment by Meakk** [comment]
> > Java interface prototype is done, but will need a complete rewrite (ideally in Kotlin) to create a good user experience (let's discuss the specs)
> 
> What does this mean ?

Basically, define the interface. Right now, I have a single OpenGL view in full screen. We may want to have a big `+` button for opening files, a previous and next button, a button to play/stop animations, etc...  
We need to sketch the user interface basically.

File: library/src/interactor_impl.cxx [hunk 1]
```diff
@@ -353,6 +353,40 @@ interactor_impl::interactor_impl(const options& options, window_impl& window, lo
...
+      break;
+  }
+}
+
+//----------------------------------------------------------------------------
+void interactor_impl::movePointer(int x, int y)
+{
+  this->Internals->VTKInteractor->SetEventPositionFlipY(x, y);
+  this->Internals->Style->OnMouseMove();
+}
```
Comment (mwestphal) [inline library/src/interactor_impl.cxx:388]: maybe a more generic API ? 

beginInteraction
moveMouse
endInteraction

?

File: library/src/interactor_impl.cxx [hunk 2]
```diff
@@ -353,6 +353,40 @@ interactor_impl::interactor_impl(const options& options, window_impl& window, lo
...
+      break;
+    default:
+      break;
+  }
+}
+
+//----------------------------------------------------------------------------
+void interactor_impl::movePointer(int x, int y)
+{
+  this->Internals->VTKInteractor->SetEventPositionFlipY(x, y);
```
Comment (mwestphal) [inline library/src/interactor_impl.cxx:386]: This seems specific to android ?

File: library/src/interactor_impl.cxx [hunk 3]
```diff
@@ -353,6 +353,40 @@ interactor_impl::interactor_impl(const options& options, window_impl& window, lo
 //----------------------------------------------------------------------------
 interactor_impl::~interactor_impl() = default;
 
+//----------------------------------------------------------------------------
+void interactor_impl::startPointer(PointerAction action, int x, int y)
```
Comment (mwestphal) [inline library/src/interactor_impl.cxx:357]: Is pointer a generic term or is it android specific ?

**Comment by Meakk** [comment]
Current status:

https://user-images.githubusercontent.com/6069752/186754316-5d9a3186-c151-49e3-89f8-4fe67fecca54.mp4

**Comment by mwestphal** [comment]
nice, the zoom is a bit janky, is it a FPS issue or interaction issue ?

**Comment by Meakk** [comment]
It's an interaction issue

**Comment by Meakk** [comment]
Android build is merged here: #382  
New required APIs here: #383  

This PR will now focus on java bindings

**Comment by Meakk** [comment]
Actually, closing this one for #390

**Pull request closed without merging**